### PR TITLE
Add Poll Rate Option to Sensors

### PR DESCRIPTION
**Pull request by meisZWFLZ**
#### Summary

This allows custom Encoders and Gyros to be created that have different poll rates than v5's.

#### Motivation

This is useful for unit testing, where time won't matter, and for other robotics applications outside of vex, like FRC.
#### Test Plan

- [ ] Test on a bot
## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/6961042424)
- via manual download: [LemLib@0.5.0-rc1-c1d8796.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1067662752.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0-rc1-c1d8796.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1067662752.zip;
pros c fetch LemLib@0.5.0-rc1-c1d8796.zip;
pros c apply LemLib@0.5.0-rc1-c1d8796;
rm LemLib@0.5.0-rc1-c1d8796.zip;
```

## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/6999984456)
- via manual download: [LemLib@0.5.0-rc1-c20c4c6.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1074173639.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0-rc1-c20c4c6.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1074173639.zip;
pros c fetch LemLib@0.5.0-rc1-c20c4c6.zip;
pros c apply LemLib@0.5.0-rc1-c20c4c6;
rm LemLib@0.5.0-rc1-c20c4c6.zip;
```

### CODE REVIEW

File: src/lemlib/odom/differentialArc.cpp [hunk 1]
```diff
@@ -174,4 +174,13 @@ void DifferentialArc::update() {
     // calculate global position
     pose += local.rotate(avgTheta);
 }
+
+int DifferentialArc::getPollRate() const {
+    int maxPollRate = 0;
+    for (auto sensor : this->horizontals) maxPollRate = std::max(maxPollRate, sensor.getPollRate());
+    for (auto sensor : this->verticals) maxPollRate = std::max(maxPollRate, sensor.getPollRate());
+    for (auto sensor : this->gyros) maxPollRate = std::max(maxPollRate, sensor->getPollRate());
+    for (auto sensor : this->drivetrain) maxPollRate = std::max(maxPollRate, sensor.getPollRate());
```
Comment (liam-teale) [inline src/lemlib/odom/differentialArc.cpp]:
```suggestion
    for (auto& sensor : this->horizontals) maxPollRate = std::max(maxPollRate, sensor.getPollRate());
    for (auto& sensor : this->verticals) maxPollRate = std::max(maxPollRate, sensor.getPollRate());
    for (auto& sensor : this->gyros) maxPollRate = std::max(maxPollRate, sensor->getPollRate());
    for (auto& sensor : this->drivetrain) maxPollRate = std::max(maxPollRate, sensor.getPollRate());
```
I may be wrong here, but shouldn't we copy by reference here?

File: build/googletest-src
Comment (liam-teale) [inline build/googletest-src:1]: i don't think this is supposed to be here

**Review by liam-teale — CHANGES_REQUESTED** [review]
Pretty good overall. Make the changes and I'll merge upon testing

File: src/lemlib/odom/differentialArc.cpp (hunk 1 above)
Comment (meisZWFLZ) [reply src/lemlib/odom/differentialArc.cpp]: Don't know, but it doesn't throw me a compile error

**Review by liam-teale — APPROVED** [review]
LGTM. Will merge upon testing

**Comment by liam-teale** [comment]
Bump, merge conflicts need resolution

**Comment by liam-teale** [comment]
closing as refactor has been closed

**Pull request closed without merging**